### ISSUE TITLE
Android game black screen / full color screen on Galaxy S6 (64-bit, arm64-v8a)

### ISSUE THREAD

**Issue by LinusNeuman** [mono/MonoGame#4262]
When launching my android game (Kontur on Google Play) on a 64-bit phone, the game will get a black screen, which will turn into green. The music will still play, but the rendering will just render a black or green screen.
This is a problem exclusive to 64-bit. I have the latest version of Xamarin. I have enabled support for all architectures.
I do not have a 64-bit phone myself so I can't debug it unfortunately, but my game artist (in another city unfortunately) has reported this error.

**Comment by LinusNeuman** [comment]
Forgot mentioning that I use the inbuilt version of MonoGame that can be "checked" when installing Visual Studio 2015 Community, does it have to do with that version never getting any updates or something like that?

**Comment by danzel** [comment]
For now you could just not build a 64 bit version and those devices can use the 32 bit one AFAIK.

Sound (not music) will crash on 64 bit as we don't have a built openal for these platforms. I have a feeling we have an existing issue about this if you have a search.

Not sure why the rendering is all broken though!

**Comment by LinusNeuman** [comment]
Sound is working though, no issues there. Only rendering that's broken.
Anyway, how do I set project properties so that it does not allow for 64-bit and only allows 64-bit architectures to run the 32-bit one?

**Comment by LinusNeuman** [comment]
Secondly, is there a way to emulate a 64-bit architecture phone? I have no so I can not get access to the catlog to debug why the rendering is weird on 64-bit version of the game.
All visual studio android emulators are x86 as far as i know.

**Comment by KonajuGames** [comment]
If sound is working then you are already building the 32-bit only version.  64-bit builds are enabled in the Android Build options in the project properties.

**Comment by KonajuGames** [comment]
What model phone is it?

**Comment by LinusNeuman** [comment]
My settings are set to use all available architectures, such as arm-v8a and x86_64, and target platform is "any", and prefer 32-bit is greyed out.
The phone that gets this was a Galaxy S6 Edge, the bigger one with an edge screen. It has armv8 if I recall correctly.

**Comment by KonajuGames** [comment]
It shouldn't work with 64-bit enabled because we do not supply 64-bit binaries for the openal-soft libraries yet.  Try disabling the 64-bit option and rebuilding to see if that has any effect.

**Comment by LinusNeuman** [comment]
Rebuilding with only support for x86 architectures did not work.

**Comment by LinusNeuman** [comment]
I still have problems with this.
I have the latest version of Xamarin and I have tried all the different build options. However, it doesn't matter how I do it. My game artist says that it gets a full black/color screen everytime if he tests my beta version. It has not been like this since he got his S6, and his S6 can play the release version of our game from July.
I really want to be able to release my patch that I have been working on since July, but 64-bit architectures seem to be unable to render as they should. Maybe a shader issue? Shadermodel? I don't know.
Sadly, I do not have a S6 or 64-bit architecture phone to test with, so I have no logs.
And there is no emulators for 64-bit unfortunately.
It feels hopeless.

**Comment by LinusNeuman** [comment]
@KonajuGames Just "pinging" you in so that I make sure you have seen this.

**Comment by KonajuGames** [comment]
I've seen it, and I'm in the same boat as you with no idea what might be causing it and no access to a Galaxy S6.  I've got a 64-bit ARM board with Android arriving in May (from a Kickstarter campaign), but that doesn't help us now.

**Comment by LinusNeuman** [comment]
That's good to hear, that you are aware of it.
I hope we will solve it as soon as possible, even if that Android in May not really helps us now.
Thanks.

**Comment by LinusNeuman** [comment]
Any news on this?

**Comment by KonajuGames** [comment]
I've got the board (arrived in late June), but haven't got Android on it yet.  I was hoping to get the case soon so I didn't have loose bits lying around, but I'll see if I can get it up and running to see if it suffers the same issue.

**Comment by LinusNeuman** [comment]
Any news?

**Comment by KonajuGames** [comment]
I'm wondering if it has anything to do with this [Mono bug report](http://www.mono-project.com/news/2016/09/12/arm64-icache/) about ARM's big.LITTLE architecture.

**Comment by LinusNeuman** [comment]
That's possible. Do you think that my bloom shader can be the thief?

**Comment by KonajuGames** [comment]
It's a CPU problem, not a shader problem.

**Comment by mrhelmut** [comment]
I'm jumping in because I'm encountering this. I have a game to ship for a client so I'm up to spend some time on this.

**Comment by mrhelmut** [comment]
The problem is still present on the latest MonoGame and Xamarin.Android 7.4.

**Comment by mrhelmut** [comment]
As far as I have tested, this seems strongly related to Exynos 64bit CPU and to the report @KonajuGames pointed: http://www.mono-project.com/news/2016/09/12/arm64-icache/

Unfortunately it looks like Mono doesn't have a 100% robust fix for this and that the problem "can rarely happen" but any MonoGame app has 100% repro chance at launch.

Disabling the ARMv8-A ABI has no effect on this problem.

This means that any recent Samsung Galaxy device won't run MonoGame application, which a huge bummer.

Related devices:

- Samsung Galaxy S6
- Samsung Galaxy S6 edge
- Samsung Galaxy S6 edge+
- Samsung Galaxy S6 Active
- Samsung Galaxy Note 4
- Samsung Galaxy Note 5
- Samsung Galaxy Tab S2
- Samsung Galaxy S5 Neo
- Samsung Galaxy S5 New
- Samsung Galaxy A3
- Samsung Galaxy A5
- Samsung Galaxy A7
- Samsung Galaxy A8
- Samsung Galaxy J7
- Samsung Galaxy S7
- Samsung Galaxy S7 edge
- Samsung Galaxy Note7
- Samsung Galaxy Note8
- Samsung Galaxy S8
- Samsung Galaxy S8+
- Samsung Galaxy View
- Meizu Pro 5
- Meizu Pro 6 Plus
- Fasetto LINK

**Comment by LinusNeuman** [comment]
@mrhelmut OnePlus 5 does not work either.

**Comment by tomspilman** [comment]
> Unfortunately it looks like Mono doesn't have a 100% robust fix for this 

@mrhelmut - So this is a Mono on Android bug and not a MonoGame bug correct?


> MonoGame app has 100% repro chance at launch.

Is there something we can change in MonoGame's launch process to avoid the issue?

Why does it only affect MonoGame and not all C# apps on Android?  What are we doing differently that triggers this more frequently on MG?

**Comment by cra0zy** [comment]
I don't think this has anything to do with the mono bug, the mono bug siegfaulted, while in this case we just don't get graphics... actually the black screen while the game is still running in the background is exactly the problem I was getting with #5410

**Comment by mrhelmut** [comment]
I'm having a few Galaxy devices sent out to me, I'll double check this.
Like @cra0zy pointed, I believe I'm running into a different issue than the one described here: I'm running into crashes on those devices, it's not a black screen problem.

**Comment by mrhelmut** [comment]
Soooo, I've triple checked with several devices and versions on a full scale game.

Latest MonoGame develop and latest Xamarin.Android (7.4.0.21), without arm64-v8a, and everything is fine on Galaxy S6, S7, S8, and A5 (all of them with 64bit Exynos CPU).

As long as arm64-v8a ABI is off.

I believe that settles the issue.

Sorry for making a fuss out of this. ^^'
